### PR TITLE
doc updates

### PR DESCRIPTION
**Pull request by nateprewitt**
Various doc updates for clarity. Some of these may not match @kennethreitz's aesthetic sensibilities, which may require an inversion of the proposed change. I was mainly trying to make things consistent, input on preferences would be greatly appreciated.

Some notes:
* I think at least the faq.rst changes *need* to be merged because there is still live documentation stating that Requests supports 3.1 and 3.2. I can open a separate PR if we want those changes quicker.

* What was ScraperWiki, is now found at quickcode.io and appears to either be a paid or account-walled service. Wayback Machine is showing the site as a free service when the documentation was created, so I'm not sure if Kenneth still wants this included. At the very least, the URL should be updated.

* ~~Does Requests support pypy 2.2 officially? faq.rst says yes, but it's other versions were wrong.~~

* While I realize it's common to use `class.method` as a pattern when describing code, I felt like `r.json` is ambiguous when referenced next to `r.text`. Calling exactly `r.json` won't function as described in the provided `ValueError` example. Explicitly defining the call as `r.json()` reduces the chance of misinterpretation.

* api.rst has a pared down set of Exceptions and classes. I assume this is to keep extraneous information from bloating the page to a point of being unhelpful. I do think it may be worth including `BaseAdapter` for easy reference since anyone looking into a custom *non-HTTP* Transport Adapter will likely need to start there.

### CODE REVIEW

File: docs/dev/todo.rst [hunk 1]
```diff
@@ -42,6 +42,7 @@ Requests currently supports the following versions of Python:
 - Python 3.4
 - Python 3.5
 - PyPy 1.9
+- PyPy 2.2
```
Comment (Lukasa) [inline docs/dev/todo.rst]: I think there is no value in being specific about PyPy versions: we just support PyPy.

File: docs/user/advanced.rst [hunk 2]
```diff
@@ -940,7 +937,7 @@ If the remote server is very slow, you can tell Requests to wait forever for
 a response, by passing None as a timeout value and then retrieving a cup of
 coffee.
 
-.. code-block:: python
+::
```
Comment (Lukasa) [inline docs/user/advanced.rst:941]: Why was this removed?

Comment (nateprewitt) [inline docs/user/advanced.rst]: :: produces the same output and is used everywhere else in the documentation. I couldn't find any additional HTML markup generated by `.. code-block:: python` v.s. `::` and the [documentation](http://www.sphinx-doc.org/en/stable/markup/code.html) states it's only necessary when displaying a lanuage not in the default config (which is python for Requests). 

It looks like this all came from the same commit from Kevin though, so he may have a reason for this that I'm missing.

Comment (nateprewitt) [inline docs/user/advanced.rst:941]: :: produces the same output and is used everywhere else in the documentation. I couldn't find any additional HTML markup generated by `.. code-block:: python` v.s. `::` and the [documentation](http://www.sphinx-doc.org/en/stable/markup/code.html) states it's only necessary when displaying a language not in the default config (which is python for Requests). 

It looks like this all came from the same commit from Kevin though, so he may have a reason for this that I'm missing.

File: docs/dev/todo.rst (hunk 1 above)
Comment (nateprewitt) [inline docs/dev/todo.rst]: Great, I'll update that.

File: docs/user/advanced.rst (hunk 2 above)
Comment (nateprewitt) [inline docs/user/advanced.rst:941]: One more note on this. It looks like this does affect the GitHub highlighting but not Sphinx (Pygments).

The two previous code snippets above this one do not currently display with any syntax highlighting, but this one does. I don't have a strong opinion on which direction to go with this, but I do think it looks a little odd if highlighting isn't uniform across grouped examples.

**Comment by sigmavirus24** [comment]
@nateprewitt is this related to #2228?

**Comment by nateprewitt** [comment]
@sigmavirus24 It wasn't originally, I found a few minor hiccups making the `.json` change and then did a pass through the docs from there. I realize this is kind of a mess of unrelated doc changes so I can break it up into different commits or PRs if needed.

It looks like most of the discussion in #2228 is about timeouts documentation. Was there something low hanging you wanted to see integrated?

**Comment by sigmavirus24** [comment]
> Was there something low hanging you wanted to see integrated?

No because the last person to comment on that thread wants to contribute and I wanted to make sure we gave them the first chance. If this is unrelated that's fine.

**Comment by nateprewitt** [comment]
Sorry, I missed that. No, none of these changes touch the timeouts docs, so I think all is well.

**Comment by kennethreitz** [comment]
These are all fanatastic changes, and ones I would have made myself, given an infinite amount of time :)

Thank you very much!
